bluepill/tests/BPIntegrationTests.m

```diff
@@ -33,15 +33,17 @@ - (void)setUp {
     self.config.appBundlePath = hostApplicationPath;
     self.config.stuckTimeout = @80;
     self.config.xcodePath = [BPUtils runShell:@"/usr/bin/xcode-select -print-path"];
-    self.config.runtime = @BP_DEFAULT_RUNTIME;
     self.config.repeatTestsCount = @1;
     self.config.errorRetriesCount = @0;
     self.config.failureTolerance = @0;
-    self.config.deviceType = @BP_DEFAULT_DEVICE_TYPE;
     self.config.headlessMode = YES;
     [BPUtils enableDebugOutput:![BPUtils isBuildScript]];
     [BPUtils quietMode:[BPUtils isBuildScript]];
     self.config.quiet = [BPUtils isBuildScript];
+    self.config.deviceType = @BP_DEFAULT_DEVICE_TYPE;
+    self.config.runtime = @BP_DEFAULT_RUNTIME;
+    // need to validate the configuration to fill in simDevice and simRuntime
```

Comment:
Maybe factor out the code that fills those in and call _that_ code from both `validateConfigWithError` and here?

Reply:
That's a good point. Thanks for pointing out.

Also, one wouldn't expect that a function like **validateConfig** would create side-effects like filling in some config variables. So, I completely detached the functionality into another action-oriented function which reads *fillSimDeviceTypeAndSimRuntime*.